src/problem_bank_scripts/problem_bank_scripts.py

```diff
@@ -80,6 +80,11 @@ def defdict_to_dict(defdict, finaldict):
                     v[k2] = dict(sorted(v2.items(), key=lambda i: i[0]))
                 
             finaldict[k] = v
+        elif hasattr(v, "dtype"):
```

Comment:
Can we add a comment for why this happens so we can have a hope of debugging if anything goes wrong?

```suggestion
        # Somewhere around Aug 2024, something changed in how objects were serialized and anything stored as a `numpy` object wasn't correctly parsed by `pyyaml`
        # See here for our history on this: (https://moosvilab.ok.ubc.ca/moosvilab/pl/zfp63dh6hjy5zdkzemzmtzqffr)
        # And the jupyterbook PR where this was also addressed: https://github.com/jupyter-book/jupyter-book/pull/2167

        elif hasattr(v, "dtype"):
```